### PR TITLE
docs: document how flags can be toggled in tests

### PR DESCRIPTION
**Pull request by dimaqq**
Document how flags can be toggled in tests
https://github.com/cfpb/django-flags/issues/121

### CODE REVIEW

**Comment by dimaqq** [comment]
@willbarton I think a maintainer needs to do something for GHA to run 🙇🏻

File: docs/testing.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    @override_settings(FLAGS={"MY_FLAG": [("boolean", False)]})
+    def test_flag_disabled(self):
+        # Do the thing that requires the flag to be disabled
+```
+
+## Advanced
+
+If for some reason you've completely disabled the `flags.sources.SettingsFlagSource` and are only allow database flags, you can either use `override_settings` to change `FLAG_SOURCES` for the tests, or if you really want to use the database, you can create a `FlagStateobject` along the same lines with a boolean condition that is `True` to test the flag-enabled code path, and then create one with a boolean condition that is` False` to test the not-enabled path.
+
+It could look something like this:
```
Comment (willbarton) [inline docs/testing.md:22]: I think you could simplify this by just presenting `FlagState` objects as an alternative to settings. Something like:

```suggestion
Alternatively, you can create a `FlagState` object along the same lines with a boolean condition that is `True` to test the flag-enabled code path, and then create one with a boolean condition that is` False` to test the not-enabled path.
```

With the example following.

**Review by willbarton — COMMENTED** [review]
Thanks, @dimaqq! 

How would you feel about simplifying the language and structure a bit and adding this to [`usage.md`](https://github.com/cfpb/django-flags/blob/main/docs/usage.md) instead of its of own file? I think this is important to call out as general usage, because folks should be writing tests! I have another suggestion about to simplify it below.

If you don't want to put this in `usage.md`, then the new file will need to be added to [`mkdocs.yml`](https://github.com/cfpb/django-flags/blob/main/mkdocs.yml) to appear in the documentation.

**Comment by dimaqq** [comment]
Makes sense, I've committed your suggestion.

**Comment by dimaqq** [comment]
Re: usage, I guess the first block, the simple case could be moved to usage 🤔

**Comment by dimaqq** [comment]
suggestions have been added, thank you :)

**Comment by dimaqq** [comment]
@willbarton wdyt, is it mergeable now?
It seems that only maintainers can merge PRs in this repo, I certainly am not allowed to.

**Comment by willbarton** [comment]
@dimaqq Correct, only maintainers can merge PRs. That's kind of basic PR security ;)

Yes, it looks good. I'll get it merged!

**Pull request merged**